### PR TITLE
Change default secret repository to cipher-text.properties

### PR DESCRIPTION
**Pull request by sachithKay**

### CODE REVIEW

File: components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java [hunk 1]
```diff
@@ -29,13 +29,16 @@
...
 	private FileSecretRepository fileSecretRepository = new FileSecretRepository();
 	private EnvironmentSecretRepository environmentSecretRepository =  new EnvironmentSecretRepository();
 
 	SecretCipherHander(org.apache.synapse.MessageContext synCtx) {
-		super();
-		parentRepository.setSynCtx(synCtx);
+		this();
+	}
+
+	SecretCipherHander() {
```
[the remark this answers is not in the document]
Comment (sachithKay) [reply components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java:40]: We don't need to pass the synapse message context anymore, as we are not using the Registry based repositorty. Hence this should be the default constructor.

File: distribution/src/conf/ciphertool.sh [hunk 2]
```diff
@@ -123,4 +123,4 @@ if $cygwin; then
 fi
 
 # ----- Execute The Requested Command -----------------------------------------
-$JAVA_HOME/bin/java -Dcarbon.home="$CARBON_HOME" -Dcarbon.config.dir.path="$CARBON_HOME"/conf -classpath "$CARBON_CLASSPATH" org.wso2.ciphertool.CipherTool $*
+$JAVA_HOME/bin/java -Dcarbon.home="$CARBON_HOME" -Dcarbon.config.dir.path="$CARBON_HOME"/conf -Dconfigure -Dorg.wso2.CipherTransformation="RSA/ECB/OAEPwithSHA1andMGF1Padding" -classpath "$CARBON_CLASSPATH" org.wso2.ciphertool.CipherTool $*
```
[the remark this answers is not in the document]
Comment (sachithKay) [reply distribution/src/conf/ciphertool.sh]: done

File: components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (Nirothipan) [reply components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java:40]: Yes , in that case we can remove the constructor in line 36.

[the remark this answers is not in the document]
Comment (sachithKay) [reply components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java:40]: shall we keep it for backward compatibility? WDYT?

File: components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java [hunk 3]
```diff
@@ -29,15 +29,10 @@
 	private static Log log = LogFactory.getLog(SecretCipherHander.class);
 
 	/* Root Secret Repository */
-	private RegistrySecretRepository parentRepository = new RegistrySecretRepository();
+	private CiphertextRepository parentRepository = CiphertextRepository.getInstance();
```
Comment (sdkottegoda) [inline components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java:32]: Is it possible to initialize an object and use it here? Or is there a specifc reason to use a singleton?

Comment (sachithKay) [reply components/mediation/security/org.wso2.micro.integrator.mediation.security/src/main/java/org/wso2/micro/integrator/mediation/security/vault/SecretCipherHander.java:32]: Ideally we should not be creating secret repository objects for each and every vault:lookup. This is to limit repository object creation.

**Pull request merged**